test/integration/deployment/deployment_test.go

```diff
@@ -689,20 +689,19 @@ func TestOverlappingDeployments(t *testing.T) {
 		t.Fatalf("expected 2 replicasets to be found, but found %d", len(rss))
 	}
 
-	var mapping map[types.UID]*v1beta1.ReplicaSet
+	m := make(map[types.UID]v1beta1.ReplicaSet)
 	for _, rs := range rss {
 		controller := metav1.GetControllerOf(&rs)
 		if controller == nil {
 			t.Fatalf("replicaset %q is not owned by any deployment", rs.Name)
 		}
-
 		// Update [controllerRef's UID: replicaset] mapping for future usage
-		mapping[controller.UID] = &rs
+		m[controller.UID] = rs
```

Comment:
I spent whole day debugging the test due to this reason. When populating Golang map and slice in a loop, we should store the object itself rather than pointer. Otherwise, the pointer will be updated in next iteration of the loop. For example, in this code block, all the values in the map will be updated to last replicaset.

Reply:
That's a common gotcha in golang -- go **reuses the same memory location** for every iteration, so you should never let the address of loop variable escape the loop.

Reply:
@janetkuo Thank you for pointing it out. I believe it is unique for Golang - I am not able to recall whether C++ has something like this, but I am not sure about this as I mostly worked on Java and JavaScript development.